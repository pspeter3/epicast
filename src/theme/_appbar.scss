@import "./animation";
@import "./elevation";
@import "./sizing";
@import "./spacing";
@import "./typography";
@import "./variables";

$appbar__spacing: 8 * $unit;

.appbar {
    @include elevation-medium();
    background-color: $surface;
    color: $white;
    display: flex;
    justify-content: space-between;
    left: 0;
    padding: $subunit;
    position: fixed;
    right: 0;
    top: 0;
}

.appbar__navigation {
    @include sizing-row();
    display: flex;
}

.appbar__title {
    @include title();
    @include spacing-target();
    margin: 0;
}

.appbar__action {
    @include spacing-target();
    border-radius: 50%;
    color: $white;
    transition: background-color $duration-simple $easing-standard,
        color $duration-simple $easing-standard;

    &:focus,
    &:hover {
        background-color: $highlight;
        color: $primary;
    }
}

#root {
    padding: $appbar__spacing $subunit 0 * $unit;
}

@supports (padding: unquote("max(0px)")) {
    .appbar,
    #root {
        padding-left: unquote("max(#{$subunit}, env(safe-area-inset-left))");
        padding-right: unquote("max(#{$subunit}, env(safe-area-inset-right))");
    }

    .appbar {
        padding-top: unquote("max(#{$subunit}, env(safe-area-inset-top))");
    }

    #root {
        // SCSS cannot handle calc and env so we save it as a CSS variable
        --notch: env(safe-area-inset-top);
        padding-top: unquote("calc(#{$appbar__spacing} + var(--notch))");
        padding-bottom: unquote("env(safe-area-inset-bottom)");
    }
}
