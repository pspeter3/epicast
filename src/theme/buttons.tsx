import * as React from "react";

export const IconButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="icon-button" type="button" {...props} />;
};

export const SubHeaderButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="subheader-button" type="button" {...props} />;
};

export const NumericSubHeaderButton: React.SFC<
    React.ButtonHTMLAttributes<HTMLButtonElement>
> = props => {
    return <button className="subheader-button numeric" type="button" {...props} />;
};

export const PrimarySubHeaderButton: React.SFC<
    React.ButtonHTMLAttributes<HTMLButtonElement>
> = props => {
    return <button className="subheader-button primary" type="button" {...props} />;
};

export const BottomButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="bottom-button" type="button" {...props} />;
};
