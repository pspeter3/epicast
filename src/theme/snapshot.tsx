import * as React from "react";
import TestRenderer from "react-test-renderer";

export const snapshot = (name: string, SFC: React.SFC<{}>): void => {
    describe(name, () => {
        it(`should match snapshot`, () => {
            const component = TestRenderer.create(<SFC>Snapshot</SFC>);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
};
