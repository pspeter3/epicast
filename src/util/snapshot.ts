import { shallow } from "enzyme";
import * as React from "react";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        expect(shallow(factory())).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
