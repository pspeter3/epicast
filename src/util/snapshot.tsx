import * as React from "react";
import TestRenderer from "react-test-renderer";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        const component = TestRenderer.create(factory());
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
