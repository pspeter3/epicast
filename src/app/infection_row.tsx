import * as React from "react";
import { classNames } from "../theme/css";
import { NumericInput } from "../theme/numeric_input";
import { Display, JustifyContent, Padding } from "../theme/tailwind";

export interface Props {
    city: string;
    value: number;
    onChange: (city: string, value: number) => void;
}

export class InfectionRow extends React.PureComponent<Props, {}> {
    public static displayName = "InfectionRow";

    public render() {
        const { city, value } = this.props;
        return (
            <div className={classNames(Display.Flex, JustifyContent.Between)}>
                <label className={classNames(Padding.A3)}>{city}</label>
                <NumericInput value={value} onChange={this._onChange} />
            </div>
        );
    }

    private _onChange = (value: number): void => {
        this.props.onChange(this.props.city, value);
    };
}
