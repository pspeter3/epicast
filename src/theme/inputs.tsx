import * as React from "react";
import { IconButton } from "./buttons";
import { CheckCircleIcon, CircleIcon } from "./icons";

export const NumericInput: React.SFC<React.InputHTMLAttributes<HTMLInputElement>> = props => {
    return <input className="numeric-input" type="number" {...props} />;
};

export const Checkbox: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    const checked = props["aria-checked"];
    return (
        <IconButton role="checkbox" aria-checked={checked} {...props}>
            {checked ? <CheckCircleIcon /> : <CircleIcon />}
        </IconButton>
    );
};
