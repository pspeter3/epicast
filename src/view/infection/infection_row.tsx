import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { TargetIcon } from "../../theme/icons";
import { Checkbox } from "../../theme/inputs";
import { Row } from "../../theme/layout";
import { Text } from "../../theme/typography";

export interface Props {
    city: string;
    checked?: boolean;
    onToggle: (city: string) => void;
    onEpidemic?: (city: string) => void;
}

export class InfectionRow extends React.PureComponent<Props, {}> {
    public render() {
        const { city, checked, onEpidemic } = this.props;
        return (
            <Row>
                <Checkbox aria-checked={checked} onClick={this._onToggle} />
                <Text>{city}</Text>
                {onEpidemic && (
                    <IconButton onClick={this._onEpidemic}>
                        <TargetIcon />
                    </IconButton>
                )}
            </Row>
        );
    }

    private _onToggle = () => {
        this.props.onToggle(this.props.city);
    };

    private _onEpidemic = () => {
        this.props.onEpidemic!(this.props.city);
    };
}
