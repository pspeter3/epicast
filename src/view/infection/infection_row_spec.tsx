import { shallow } from "enzyme";
import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { Checkbox } from "../../theme/inputs";
import { snapshot } from "../../util/snapshot";
import { InfectionRow } from "./infection_row";

describe("InfectionRow", () => {
    describe("snapshots", () => {
        it(
            "should match default snapshot",
            snapshot(() => <InfectionRow city="Test" onToggle={jest.fn()} />),
        );
        it(
            "should match checked snapshot",
            snapshot(() => <InfectionRow city="Test" onToggle={jest.fn()} />),
        );
        it(
            "should match epidemic snapshot",
            snapshot(() => (
                <InfectionRow
                    city="Test"
                    checked={false}
                    onToggle={jest.fn()}
                    onEpidemic={jest.fn()}
                />
            )),
        );
    });

    describe("handlers", () => {
        it("should trigger change", () => {
            const city = "San Francisco";
            const onToggle = jest.fn();
            const tree = shallow(<InfectionRow city={city} onToggle={onToggle} />);
            tree.find(Checkbox).simulate("click");
            expect(onToggle).toHaveBeenCalledWith(city);
        });

        it("should trigger epidemic", () => {
            const city = "San Francisco";
            const onEpidemic = jest.fn();
            const tree = shallow(
                <InfectionRow city={city} onToggle={jest.fn()} onEpidemic={onEpidemic} />,
            );
            tree.find(IconButton).simulate("click");
            expect(onEpidemic).toHaveBeenCalledWith(city);
        });
    });
});
