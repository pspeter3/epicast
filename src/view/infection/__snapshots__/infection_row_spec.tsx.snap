// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfectionRow snapshots should match checked snapshot 1`] = `
<Component>
  <Component
    onClick={[Function]}
  />
  <Component>
    Test
  </Component>
</Component>
`;

exports[`InfectionRow snapshots should match default snapshot 1`] = `
<Component>
  <Component
    onClick={[Function]}
  />
  <Component>
    Test
  </Component>
</Component>
`;

exports[`InfectionRow snapshots should match epidemic snapshot 1`] = `
<Component>
  <Component
    aria-checked={false}
    onClick={[Function]}
  />
  <Component>
    Test
  </Component>
  <Component
    onClick={[Function]}
  >
    <Component />
  </Component>
</Component>
`;
